h2o-core/src/main/java/water/Scope.java

```diff
@@ -86,14 +86,17 @@ public static Frame track(Frame... frames) {
     for (Frame fr : frames) {
       Scope scope = _scope.get();
       assert scope != null;
-      track_impl(scope, fr._key);
+      if (fr._key != null) {
+        track_impl(scope, fr._key);
+      }
       for (Vec vec : fr.vecs())
         track_impl(scope, vec._key);
     }
     return frames[0];
   }
 
   static private void track_impl(Scope scope, Key key) {
+    assert key != null;
```

Comment:
Why do you need this assert here? Wouldn't it make more sense to check if the key is null and if it is simply don't do anything?

Some cases are not handled. You ignore Frames with 'null' keys but not Vecs with null keys. Why this inconsistency?

Reply:
This assert started as a leftover. I noticed there should be some more checks at the beginning of the track_impl method, but did not really implement them afterwards, as it was not directly related to the issue. Let me address this shortly.